Cython/Includes/libc/threads.pxd

```diff
@@ -77,8 +83,150 @@ cdef extern from "<threads.h>" nogil:
         pass
     enum:
         TSS_DTOR_ITERATIONS
-    ctypedef void (*tss_dtor_t)(void*) nogil noexcept
+    ctypedef void (*tss_dtor_t)(void*) noexcept nogil
     int tss_create(tss_t* key, tss_dtor_t destructor)
     void *tss_get(tss_t key)
     int tss_set(tss_t key, void* val)
     void tss_delete(tss_t key)
+
+cdef extern from *:
+    """
+    static int __pyx_libc_threads_has_gil(PyGILState_STATE *gil_state) {
+        #if CYTHON_COMPILING_IN_LIMITED_API
+            /* There is no way to know if we have the GIL. Therefore the only
```

Comment:
I've come across evidence that says this may not actually be true. And that we can probably avoid this mess. But I need to investigate it first.